File: german-telework/service/tasks/NetworkPacket.java
public class NetworkPacket {

    User user;
    Task task;

    public NetworkPacket (User user, Task task) {
        this.user = user;
        this.task = task;
    }

    public static NetworkPacket deserialize (String data) throws WrongTargetException, PacketFormatException {
        String[] parts = data.split("\\|\\|\\|");
        if (!(parts[0].equals("0"))){
            throw new WrongTargetException("This packet is not for me!");
        }
        String user_ser = parts[1];
        String task_ser = parts[2];
        User user = User.deserialize(user_ser);
        Task task = Task.deserialize(task_ser);
        return new NetworkPacket(user, task);
    }

    public User getUser(){
        return this.user;
    }

    public Task getTask(){
        return this.task;
    }

    public static String reply(User user, Task task){
        user.addTask(task.serialize());
        String res = "4|||";
        res += user.serialize();
        return res.strip();
    }
}

File: german-telework/service/holiday/Holiday/NetworkPacket.cs
public class NetworkPacket {

    public User user {get; }
    public Leave leave {get; }

    public NetworkPacket(User user, Leave leave){
        this.user = user;
        this.leave = leave;
    }

    public static NetworkPacket Deserialize(String packet){
        packet = packet.TrimEnd('\n');
        String[] parts = packet.Split("|||");
        User user = User.Deserialize(parts[1]);
        Leave leave = Leave.Deserialize(parts[2]);
        if (user != null){
            return new NetworkPacket(user, leave);
        } else {
            return null;
        }
    }

    public String Serialize() {
        String res = "0|||";
        res += user.Serialize();
        return res;
    }

    public static String reply(User user, Leave leave) {
        if (leave != null){
            user.AddLeave(leave.Serialize());
        }
        String res = "4|||";
        res += user.Serialize();
        return res;
    }

}
